### DISCUSSION TITLE
batch remove black background.

### DISCUSSION THREAD

**Discussion by Speedwagen** [ImageMagick/ImageMagick#6846] [Digital Image Processing]
The original file was a .bmp that has no alpha channel but uses black to act as transparency.
I have now converted all files into .png format and now I am looking to batch-remove the black background.
The files are from a video game with open-source components so I am hesitant to upload the textures so I am uploading a generic file.
[particle.zip](https://github.com/ImageMagick/ImageMagick/files/13249809/particle.zip)

**Comment by snibgo** [comment]
If you want to make transparent all pixels that are exactly black:
```
magick smoke.png -transparent Black out.png
```
You can put that in a shell `for` loop to process all files in a directory, or use `magick mogrify`.

**Reply by Speedwagen** [reply]
I think thats where the issue is. I can try to fuzz it but that might cause artifacting.

**Comment by snibgo** [comment]
Perhaps you don't want to make transparent all pixels that are exactly black. Can you say what you do want? Can you provide a sample image before and after the transformation that you do want?

**Reply by Speedwagen** [reply]
yes exactly. It's hard to provide an example but I think I may have found a file that had correctly been generated.
I will provide 2 sets of images. one where the file has been corrupted and another where it is correct.
![5AC4A37883971DAEworking alpha](https://github.com/ImageMagick/ImageMagick/assets/110987088/b7c016b0-83ee-45e7-ac20-79a4916b3581) Working Alpha
![5AC4A37883971DAEno alpha](https://github.com/ImageMagick/ImageMagick/assets/110987088/caef0cb0-3372-41f7-8347-22da5dfd19ca) No Alpha
![4DE99005C18225ECbroken alpha](https://github.com/ImageMagick/ImageMagick/assets/110987088/7be191e7-e9fa-4275-8dc3-fd1e86ffd8de)Non Working Alpha
![4DE99005C18225ECno alpha](https://github.com/ImageMagick/ImageMagick/assets/110987088/a963cba9-cf91-40ec-a78b-80aa2a5cf562) No Alpha

**Comment by snibgo** [comment]
The forum software changes the filenames. Please edit your post to explain what each image is.

**Reply by Speedwagen** [reply]
edited!

**Comment by snibgo** [comment]
In the first pair of "working alpha" and "no alpha" images, I suppose the "no alpha" is your input, and you want to a transformation that makes the "working alpha" image. Is that correct?

Pixels in "no alpha" that are exactly black are made transparent. And so are some pixels that are close to black, and some pixels that are not close to black.

Also, most non-black pixels change their colour.

So I don't know what will transform "no alpha" to "working alpha".

**Reply by Speedwagen** [reply]
yes, pretty much what you said here - "Pixels in "no alpha" that are exactly black are made transparent. And so are some pixels that are close to black, and some pixels that are not close to black."

is there a way to use mogrify with alpha to set the pixel that are black and "near to" black as the alpha channel?

**Comment by snibgo** [comment]
```
magick mogrify -fuzz 1% -transparent Black -path mog noAlp.png
```
This writes a new file to the subdirectory `mog`, making transparent all pixels that are within 1% of black. Comparing that to your "working alpha" image, it has made too many pixels transparent in some areas, and not enough in other areas.

**Reply by Speedwagen** [reply]
I tried to run this but it seems like its looking for a file called noAlp.png
I tried to change it to *.png but then cmd exploded haha

**Comment by snibgo** [comment]
You labelled an image "no alpha". I downloaded that and named it `noAlp.png`. You should use whatever filename you have. Or use `*.png` for all png files, or `n*.png`, or whatever.

EDIT: If you use `-path`, the path you name must already exist.
